dashpay/src/main/kotlin/org/dashevo/dashpay/BlockchainIdentity.kt

```diff
@@ -0,0 +1,767 @@
+                val saltedDomainHashData = saltedDomainHashes[username] as ByteArray
+                val saltedDomainHashString = "5620${saltedDomainHashData.toHexString()}"
+                for (preorderDocument in preorderDocuments) {
+                    if (preorderDocument.data["saltedDomainHash"] == saltedDomainHashString) {
+                        var usernameStatus = if (usernameStatuses.containsKey(username))
+                            usernameStatuses[username] as MutableMap<String, Any>
+                        else HashMap()
+                        usernameStatus[BLOCKCHAIN_USERNAME_STATUS] = UsernameStatus.PREORDERED
+                        usernameStatuses[username] = usernameStatus
+                        saveUsername(username, UsernameStatus.PREORDERED, null, true)
+                        usernamesLeft.remove(username)
+                    }
+                }
+            }
+            if (usernamesLeft.size > 0 && retryCount > 0) {
+                val saltedDomainHashesLeft = saltedDomainHashes.filter { usernamesLeft.containsKey(it.key) }
+                Timer("monitorBlockchainIdentityStatus", false).schedule(timerTask {
+                    val nextDelay = delayMillis * when (retryDelayType) {
+                        RetryDelayType.SLOW20 -> 5 / 4
+                        RetryDelayType.SLOW50 -> 3 / 2
+                        else -> 1
+                    }
+                    monitorForDPNSPreorderSaltedDomainHashes(
+                        saltedDomainHashesLeft,
+                        retryCount - 1,
+                        nextDelay,
+                        retryDelayType,
+                        callback
+                    )
+                }, delayMillis)
+            } else if (usernamesLeft.size > 0) {
+                val saltedDomainHashesLeft = saltedDomainHashes.filter { usernamesLeft.containsKey(it.key) }
+                callback.onTimeout(saltedDomainHashesLeft.keys.toList())
+            } else {
+                callback.onComplete(saltedDomainHashes.keys.toList())
+            }
+        } else {
+            if (retryCount > 0) {
+                Timer("monitorForDPNSPreorderSaltedDomainHashes", false).schedule(timerTask {
+                    val nextDelay = delayMillis * when (retryDelayType) {
+                        RetryDelayType.SLOW20 -> 5 / 4
+                        RetryDelayType.SLOW50 -> 3 / 2
+                        else -> 1
+                    }
+                    monitorForDPNSPreorderSaltedDomainHashes(
+                        saltedDomainHashes,
+                        retryCount - 1,
+                        nextDelay,
+                        retryDelayType,
+                        callback
+                    )
+                }, delayMillis)
+            } else {
+                callback.onTimeout(saltedDomainHashes.keys.toList())
+            }
+        }
+        //throw exception or return false
+    }
+
+    fun monitorForDPNSUsernames(
```

Comment:
Would `watchDPNSUsernames` mean the same? If so, I would suggest it as it's easier to understand.

Reply:
If you think this suggestion makes sense, the same would apply for the other methods with the `monitorFor` prefix.

Reply:
Yes, it would mean the same.  I was using the iOS as a guide, but those developers seem to like really long method names.  I will shorten these method names.

Reply:
This and related method names were shorted.  For example:

watchIdentity
watchPreorder
watchUsernames

https://github.com/dashevo/android-dashpay/pull/1/commits/52e8a22bf97fc647a62bd624d4ce49b5030a7c1c